PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/KustoQueriesWithTimeIntervalInQueryStringCheck.java

```diff
@@ -0,0 +1,258 @@
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiExpression;
+import com.intellij.psi.PsiExpressionList;
+import com.intellij.psi.PsiLiteralExpression;
+import com.intellij.psi.PsiLocalVariable;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiPolyadicExpression;
+import com.intellij.psi.PsiReferenceExpression;
+import com.intellij.psi.PsiVariable;
+import com.intellij.psi.impl.source.tree.java.PsiMethodCallExpressionImpl;
+
+import com.intellij.psi.util.PsiTreeUtil;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+/**
+ * This class is an inspection tool that checks for Kusto queries with time intervals in the query string.
+ * This approach makes queries less flexible and harder to troubleshoot.
+ * This inspection tool checks for the following anti-patterns:
+ * 1. Queries that use the "ago" function with a time interval.
+ * 2. Queries that use the "datetime" function with a specific datetime.
+ * 3. Queries that use the "now" function to get the current timestamp.
+ * 4. Queries that use the "startofday", "startofmonth", or "startofyear" functions.
+ * 5. Queries that use the "between" function with datetime values.
+ * <p>
+ * When the anti-patterns are detected as parameters of Azure client method calls, a problem is registered.
+ */
+public class KustoQueriesWithTimeIntervalInQueryStringCheck extends LocalInspectionTool {
+
+    /**
+     * This method builds the visitor for the inspection tool
+     *
+     * @param holder     ProblemsHolder is used to register problems found in the code
+     * @param isOnTheFly boolean to indicate if the inspection is done on the fly -
+     * @return PsiElementVisitor
+     */
+    @NotNull
+    @Override
+    public JavaElementVisitor buildVisitor(@NotNull ProblemsHolder holder, boolean isOnTheFly) {
+        return new KustoQueriesVisitor(holder, isOnTheFly);
+    }
+
+    /**
+     * This class defines the visitor for the inspection tool
+     * The visitor checks for Kusto queries with time intervals in the query string
+     * and registers a problem if an anti-pattern is detected
+     * To check for the anti-patterns, the visitor uses regex patterns to match the query string
+     * Processing of polyadic expressions is also done to replace the variables with their values in the query string
+     * before checking for the anti-patterns.
+     */
+    static class KustoQueriesVisitor extends JavaElementVisitor {
+
+        private final ProblemsHolder holder;
+
+        // // Define constants for string literals
+        private static final RuleConfig RULE_CONFIG;
+        private static final String ANTI_PATTERN_MESSAGE;
+        private static final Map<String, Pattern> REGEX_PATTERNS = new HashMap<>();
```

Comment:
Could this be a `List` of `Pattern` instead of `Map<String, Pattern>`, I don't see the key values being sued anywhere.

Reply:
done:)